library/src/main/java/com/datatorrent/lib/io/block/ReaderContext.java

```diff
@@ -190,11 +190,11 @@ protected Entity readEntity() throws IOException
           if (bytesRead == -1) {
             break;
           }
-          strBuffer = new String(buffer);
+          bufferStr = new String(buffer,0, bytesRead);
         }
 
-        while (posInStr < strBuffer.length()) {
-          char c = strBuffer.charAt(posInStr);
+        while (posInStr < bufferStr.length()) {
+          char c = bufferStr.charAt(posInStr);
           if (c != '\r' && c != '\n') {
             tmpBuilder.write(c);
```

Comment:
@gauravgopi123 How about creating separate JIRA for addressing this issue?
We can have in-depth discussion on the issues, alternatives, possible solutions on the new JIRA.